common/flatpak-dir.c

```diff
@@ -5358,12 +5358,13 @@ export_mime_file (int           parent_fd,
 static char *
 format_flatpak_run_args_from_run_opts (GStrv flatpak_run_args)
 {
-  GString *str = g_string_new ("");
+  GString *str;
   GStrv iter = flatpak_run_args;
 
   if (flatpak_run_args == NULL)
     return NULL;
 
+  str = g_string_new ("");
```

Comment:
How does separating the initialization from the declaration fix a memory leak?

Reply:
As we're returning NULL without freeing the GString container if flatpak_run_args is NULL.